### DISCUSSION TITLE
Need help understanding server side vs static generation docs

### DISCUSSION THREAD

**Discussion by burrack** [vercel/next.js#17284] [Help] [answered]
The [docs](https://nextjs.org/learn/basics/data-fetching/two-forms) is saying the following:

> You can use Static Generation for many types of pages, including:
> 
> Marketing pages
> Blog posts
> E-commerce product listings
> Help and documentation

E-commerce product listings is referring to the product page itself? or the page that the user is looking for products?
It makes sense to me that a product page can be generated at build time.
Usually the products page (the browse page for products) has a lot of filters that handle different api requests and probably pagination. I assume you have to do server side rendering, no?

Thanks

**Comment by jamesmosier** [comment, marked as answer]
I can't speak to what the intent of the docs was (to me it sounds just like an individual product page), but the initial load of a products page (list of products) *could* be statically generated. Then as a user filters results you could just do all of that client side. I don't think every iteration of possible filters would be needed to be statically generated (for SEO purposes).

**Reply by burrack** [reply]
Thanks @jamesmosier.
yea, what I was thinking ,
Just wanted to make sure I understood it correctly :)

**Comment by lfades** [comment]
> E-commerce product listings is referring to the product page itself? or the page that the user is looking for products?

It can perfectly be used for the product page itself, our docs only mention some examples.

> Usually the products page (the browse page for products) has a lot of filters that handle different api requests and probably pagination. I assume you have to do server side rendering, no?

Let's say you have 2 paths: `/products` and `/products/search?q=anything`. You can do the following:

- Statically generate `/products` using `getStaticProps`
- Use a fallback state for `/products/search?q=anything` and fetch the products in the browser instead of doing it at build time (because you can't possibly know all the searches your visitors can do before runtime - and you don't need SEO out of custom searches)

It's not required to use SSR in any of the mentioned cases 👍
